### PR TITLE
feat: add OnBattlegroundDesertion hook

### PR DESCRIPTION
**Pull request by 0xCiBeR**
Added OnBattlegroundDesertion hook.

Sample Lua code to test:
```lua
local function onDeserter(event, player, type)
    player:SendAreaTriggerMessage( "It works!" )
    print("it works")
end
RegisterPlayerEvent(57, onDeserter)
```

### CODE REVIEW

**Comment by 0xCiBeR** [comment]
Hey @r-o-b-o-t-o just leaving a ping here so you can check the PR when you have the time.. Thanks!

File: src/LuaEngine/PlayerHooks.cpp [hunk 1]
```diff
@@ -708,3 +708,11 @@ void Eluna::OnRemoveAura(Player* player, Aura* aura, bool isExpired)
...
     CallAllFunctions(PlayerEventBindings, key);
 }
+
+void Eluna::OnBattlegroundDesertion(Player* player, const BattlegroundDesertionType type)
+{
+    START_HOOK(PLAYER_EVENT_ON_BG_DESERTION);
+    Push(player);
+    Push(type);
+    CallAllFunctions(PlayerEventBindings, key);
+}
```
Comment (55Honey) [inline src/LuaEngine/PlayerHooks.cpp:718]: Thanks for the PR. THere's a missing new line at the end of that file. See the red circle with a line inside.

Comment (0xCiBeR) [reply src/LuaEngine/PlayerHooks.cpp]: Fixed now @55Honey

**Comment by 0xCiBeR** [comment]
Hey! Is there anything else required on my side to merge the PR? Seems tests passed fine :)

**Comment by 55Honey** [comment]
Please be patient until it's reviewed. I can't think of anything missing from a quick glance.

**Comment by r-o-b-o-t-o** [comment]
Thanks for the PR!

Sorry for taking this long, I needed to remove another player hook first. This also resulted in your event being changed from `59` to `57`. I edited your sample Lua code in your first message.

**Pull request merged**